datafusion/core/tests/optimizer_integration.rs

```diff
@@ -115,6 +117,139 @@ fn concat_ws_literals() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn anti_join_with_join_filter() -> Result<()> {
```

Comment:
why are these tests moved? Does that mean we can't plan SQL with identifiers like `test.col_int32` without an Expr planner?

Reply:
For the earlier implementation, I also moved plannig of basic SQL like column references to expr planner, we had to move these test cases from `optimizer` crate to `core`  to avoid `optimizer` crate dependecy on `functions`.

Since we have moved that out of planner, ideally i should moved the testcases back. My bad, i missed them. Thanks for catching this. I have fixed this.